x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -44,64 +83,89 @@ export async function getDocumentSources({
         ? docTypeConfig.rollupIntervals
         : [RollupInterval.OneMinute]
     ).flatMap((rollupInterval) => {
-      const searchParams = {
-        apm: {
-          sources: [
+      return {
+        documentType,
+        rollupInterval,
+        meta: {
+          checkSummaryExistsField: false,
+        },
+        before: getFilters({
+          documentType,
+          rollupInterval,
+          filters: [...kql, ...beforeRange],
+        }),
+        current: getFilters({
+          documentType,
+          rollupInterval,
+          filters: [...kql, ...currentRange],
+        }),
+      };
+    });
+  });
+
+  const sourcesToCheckWithSummary = [
+    ApmDocumentType.TransactionMetric as const,
+  ].flatMap((documentType) => {
+    const docTypeConfig = getConfigForDocumentType(documentType);
+
+    return (
+      enableContinuousRollups
+        ? docTypeConfig.rollupIntervals
+        : [RollupInterval.OneMinute]
+    ).flatMap((rollupInterval) => {
+      const summaryExistsFilter = {
+        bool: {
+          filter: [
             {
-              documentType,
-              rollupInterval,
+              exists: {
+                field: TRANSACTION_DURATION_SUMMARY,
+              },
             },
           ],
         },
-        body: {
-          track_total_hits: 1,
-          size: 0,
-          terminate_after: 1,
-        },
       };
 
       return {
         documentType,
         rollupInterval,
-        before: {
-          ...searchParams,
-          body: {
-            ...searchParams.body,
-            query: {
-              bool: {
-                filter: [...kql, ...beforeRange],
-              },
-            },
-          },
-        },
-        current: {
-          ...searchParams,
-          body: {
-            ...searchParams.body,
-            query: {
-              bool: {
-                filter: [...kql, ...currentRange],
-              },
-            },
-          },
+        meta: {
+          checkSummaryExistsField: true,
         },
+        before: getFilters({
+          documentType,
+          rollupInterval,
+          filters: [...kql, ...beforeRange, summaryExistsFilter],
+        }),
+        current: getFilters({
+          documentType,
+          rollupInterval,
+          filters: [...kql, ...currentRange, summaryExistsFilter],
+        }),
       };
     });
   });
 
-  const allSearches = sourcesToCheck.flatMap(({ before, current }) => [
+  const docSearches = sourcesToCheck.flatMap(({ before, current }) => [
     before,
     current,
   ]);
 
+  const summaryFieldSearches = sourcesToCheckWithSummary.flatMap(
```

Comment:
you can concatenate `sourcesToCheck` and `sourcesToCheckWithSummary` before flatMapping on both